packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -473,18 +473,20 @@ export class Cluster extends Resource implements ICluster {
     const asg = new autoscaling.AutoScalingGroup(this, id, {
       ...options,
       vpc: this.vpc,
-      machineImage: new EksOptimizedImage({
-        nodeType: nodeTypeForInstanceType(options.instanceType),
-        kubernetesVersion: this.version,
-      }),
+      machineImage: options.machineImageType === MachineImageType.BOTTLEROCKET ?
+        new BottleRocketImage({ cluster: this }) :
+        new EksOptimizedImage({
+          nodeType: nodeTypeForInstanceType(options.instanceType),
+          kubernetesVersion: this.version,
+        }),
       updateType: options.updateType || autoscaling.UpdateType.ROLLING_UPDATE,
       instanceType: options.instanceType,
     });
 
     this.addAutoScalingGroup(asg, {
       mapRole: options.mapRole,
-      bootstrapOptions: options.bootstrapOptions,
-      bootstrapEnabled: options.bootstrapEnabled
+      bootstrapOptions: options.machineImageType === MachineImageType.BOTTLEROCKET ? undefined : options.bootstrapOptions,
```

Comment:
Feels like if users specify `bootstrap` options with BOTTLEROCKET they should get an error, not have it ignored, no?

Reply:
Also, why can't we specify bootstrap options for BOTTLEROCKET?